pandas/tseries/tests/test_period.py

```diff
@@ -4457,6 +4451,42 @@ def test_ops_frame_period(self):
         tm.assert_frame_equal(df - df2, -exp)
 
 
+class TestAstype(tm.TestCase):
+
+    def test_periodindex(self):
+        # GH 13149, GH 13209
+        idx = PeriodIndex(['2016-05-16', 'NaT', NaT, np.NaN], freq='D')
+
+        result = idx.astype(object)
+        expected = Index([Period('2016-05-16', freq='D')] +
+                         [Period(NaT, freq='D')] * 3, dtype='object')
+        # Hack because of lack of support for Period null checking (GH12759)
+        tm.assert_index_equal(result[:1], expected[:1])
+        result_arr = np.asarray([p.ordinal for p in result], dtype=np.int64)
+        expected_arr = np.asarray([p.ordinal for p in expected],
+                                  dtype=np.int64)
+        tm.assert_numpy_array_equal(result_arr, expected_arr)
+        # TODO: When GH12759 is resolved, change the above hack to:
+        # tm.assert_index_equal(result, expected)         # now, it raises.
```

Comment:
I think I would move this class to ``tests/indexes/test_datetimelike.py`` (also you can break this into some sub-tests as its within a class). Ideally we like more short test functions (and more of them) rather than longer ones.